src/Controllers/GagePageController.ts

```diff
@@ -691,10 +699,70 @@ module StreamStats.Controllers {
                         dailyValues = 0
                     };
                     this.dailyFlow = dailyValues
-                    this.getRatingCurve();
+                    this.getNWSForecast();
                 }); 
             }
 
+            public getNWSForecast() {
+                var self = this;
+                var nwisCode = this.gage.code;
+            
+                this.$http.get('./data/gageNumberCrossWalk.json').then(function(response) {
+                    self.crossWalk = response.data;
+                    var NWScode = self.crossWalk[nwisCode];
+            
+                    if (NWScode !== undefined) {
+                        var url = "https://water.weather.gov/ahps2/hydrograph_to_xml.php?output=xml&gage=" + NWScode;
+                        const request: WiM.Services.Helpers.RequestInfo = new WiM.Services.Helpers.RequestInfo(url, true, WiM.Services.Helpers.methodType.GET, 'xml');
+            
+                        self.Execute(request).then(function(response) {
+                            const xmlDocument = new DOMParser().parseFromString(response as string, "text/xml");
+
+                            const sigStages = xmlDocument.querySelector("sigstages");
+            
+                            const action = parseFloat(sigStages.querySelector("action").textContent);
```

Comment:
If you try to open the gage page for some gages (ex. 12340000), I get an error on this line: `angular.js:15635 TypeError: Cannot read properties of null (reading 'querySelector')` and the plots won't load.